packages/basis/src/BasisLoader.ts

```diff
@@ -0,0 +1,511 @@
+import { INTERNAL_FORMATS, TYPES, MIPMAP_MODES, ALPHA_MODES, FORMATS } from '@pixi/constants';
+import { resources, BaseTexture, Texture } from '@pixi/core';
+import { Runner } from '@pixi/runner';
+import {
+    BASIS_FORMATS,
+    BASIS_FORMAT_TO_INTERNAL_FORMAT,
+    INTERNAL_FORMAT_TO_BASIS_FORMAT,
+    BASIS_FORMATS_ALPHA,
+    BasisTextureExtensions,
+    BasisBinding,
+    BASIS_FORMAT_TO_TYPE
+} from './Basis';
+import { ITranscodeResponse } from './TranscoderWorkerWrapper';
+import { TranscoderWorker } from './TranscoderWorker';
+import { ILoaderResource, LoaderResource } from '@pixi/loaders';
+
+const { CompressedTextureResource } = resources;
+
+type TranscodedResourcesArray = (Array<resources.CompressedTextureResource> | Array<resources.BufferResource>) & {
+    basisFormat: BASIS_FORMATS
+};
+
+LoaderResource.setExtensionXhrType('basis', LoaderResource.XHR_RESPONSE_TYPE.BUFFER);
+
+/**
+ * Loader plugin for handling BASIS supercompressed texture files.
+ *
+ * To use this loader, you must bind the basis_universal WebAssembly transcoder. There are two ways of
+ * doing this:
+ *
+ * 1. Adding a &lt;script&gt; tag to your HTML page to the transcoder bundle in this package, and serving
+ * the WASM binary from the same location.
+ *
+ * ```js
+ * // Copy ./node_modules/@pixi/basis/assets/basis_.wasm into your assets directory
+ * // as well, so it is served from the same folder as the JavaScript!
+ * &lt;script src="./node_modules/@pixi/basis/assets/basis_transcoder.js" /&gt;
+ * ```
+ *
+ * NOTE: `basis_transcoder.js` expects the WebAssembly binary to be named `basis_transcoder.wasm`.
+ * NOTE-2: This method supports transcoding on the main-thread. Only use this if you have 1 or 2 *.basis
+ * files.
+ *
+ * 2. Loading the transcoder source from a URL.
+ *
+ * ```js
+ * // Use this if you to use the default CDN url for @pixi/basis
+ * BasisLoader.loadTranscoder();
+ *
+ * // Use this if you want to serve the transcoder on your own
+ * BasisLoader.loadTranscoder('./basis_transcoder.js', './basis_transcoder.wasm');
+ * ```
+ *
+ * NOTE: This can only be used with web-workers.
```

Comment:
Missing `@class` and `@memberof PIXI`

http://pixijs.download/feature/compressed-texture-workflow/docs/BasisLoader.html
that should be in the `PIXI` namespace.